api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -291,6 +295,68 @@ scalar UserAssignedIdentityResourceId
     }
   ]>;
 
+/** A representation of a KeyVault Secret. */
+model KmsKey {
+  /** vaultName is the name of the keyvault that contains the secret */
+  @maxLength(255)
+  @minLength(1)
+  vaultName: string;
+
+  /** name is the name of the keyvault key used for encrypt/decrypt */
+  @maxLength(255)
+  @minLength(1)
+  name: string;
+
+  /** version contains the version of the key to use */
+  @maxLength(255)
+  @minLength(1)
+  version: string;
+}
+
+/** A KMS Key that is immutable. */
+@withVisibilityFilter(#{ all: #[Lifecycle.Create, Lifecycle.Read] })
+model NonUpdatableKmsKey {
+  ...KmsKey;
+}
+
+/** The ETCD settings and configuration options. */
+model EtcdProfile {
+  /** ETCD Data Encryption settings.
+   * If not specified platform managed keys are used.
+   */
+  dataEncryption?: EtcdDataEncryptionProfile;
+}
+
+/** The ETCD data encryption settings. */
+model EtcdDataEncryptionProfile {
+  /** Specify whether the key is customer or platform managed.
```

Comment:
We don't need to say "customer or platform managed". We can just say specify the key management mode, and then describe what is the effect of each.  In that way it is not limited to just two

Reply:
Done